### PR TITLE
Add Java JAR project model

### PR DESCRIPTION
**Pull request by gandreadis**
In order to support projects coming in JAR format, I'm adding a new class `JavaJarProject`. The class effectively only serves to hold class names - extracting the `.class` files from a JAR to store them, too, is not useful, IMO (we don't use them anywhere).

As this really didn't fit anymore in the `java-maven-project` naming scheme, I've renamed the module to `java-project` and the package from `javamaven` to `java` (with `MavenProject` being one level higher in the package structure).

### CODE REVIEW

File: modules/models/java-project/src/main/kotlin/org/cafejojo/schaapi/models/project/java/JavaJarProject.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package org.cafejojo.schaapi.models.project.java
+
+import java.io.File
+
+/**
+ * A Java project contained in a JAR file.
+ */
+@SuppressWarnings("LateinitUsage") // Refer to PR #23
+class JavaJarProject(override val classDir: File) : JavaProject {
```
Comment (FWDekker) [inline modules/models/java-project/src/main/kotlin/org/cafejojo/schaapi/models/project/java/JavaJarProject.kt]: I think it's confusing that the constructor expects `classDir`. What do you think about replacing it with a `private val jar: File` argument, and overriding the `classDir` field with a custom getter?

Comment (FWDekker) [inline modules/models/java-project/src/main/kotlin/org/cafejojo/schaapi/models/project/java/JavaJarProject.kt]: Could you `require` somewhere (in the `init`, probably) that the file at `classDir` is an accessible file?

Comment (gandreadis) [reply modules/models/java-project/src/main/kotlin/org/cafejojo/schaapi/models/project/java/JavaJarProject.kt]: Good idea! Will fix

Comment (gandreadis) [reply modules/models/java-project/src/main/kotlin/org/cafejojo/schaapi/models/project/java/JavaJarProject.kt]: I've done an 'if' check for now, for consistency (the other classes in the package do this to). At some point, we should change all of them to `require`s

File: modules/models/java-project/src/main/kotlin/org/cafejojo/schaapi/models/project/java/JavaJarProject.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    override val dependencyDir: File = classDir
+    override var classes: List<File> = listOf()
+    override var dependencies: List<File> = listOf()
+    override var classpath: String = classDir.absolutePath
+    override val projectDir: File = classDir
+
+    override lateinit var classNames: List<String>
+
+    init {
+        if (!jar.isFile) {
```
Comment (JSAbrahams) [inline modules/models/java-project/src/main/kotlin/org/cafejojo/schaapi/models/project/java/JavaJarProject.kt]: This can probably be inlined.

Comment (JSAbrahams) [reply modules/models/java-project/src/main/kotlin/org/cafejojo/schaapi/models/project/java/JavaJarProject.kt]: As discussed IRL, no.

Comment (gandreadis) [reply modules/models/java-project/src/main/kotlin/org/cafejojo/schaapi/models/project/java/JavaJarProject.kt]: Yes, but the other class does not do this. Will fix though when I do `require` for all

**Pull request merged**